autogen/header.rs

```diff
@@ -127,6 +130,20 @@ fn gen_value_enum_operand_kind(grammar: &structs::OperandKind) -> TokenStream {
                 }
             }
         }
+
+        impl num_traits::FromPrimitive for #kind {
+            #[allow(trivial_numeric_casts)]
+            fn from_i64(n: i64) -> Option<Self> {
+                match n as u32 {
```

Comment:
can we have this refactored as
```rust
Some(match n as u32 {
   ...
   _ => return None
})
```

Reply:
That's give the match statement incompatible types, right? Eg, the enum type and `None`.

Reply:
no, it's fine. The `return` isn't affected by a wrapping `Some`

Reply:
Do you want the keep the match arms as `21 => Some(....)` then? And if so, what's the point of the extra `Some(match...`?

Reply:
no, of course, the arms would become `21 => Enum::SomeVariant`